File: Stack/stack.js
const input = require("prompt-sync")();

class stack{
    constructor(size=5){
        this.size=size;
        this.array = new Array(size);
        this.top=-1;
        console.log("stack created of size", size);
    }

    push(value){
        if(this.top==this.size-1){
            console.log("stack is overflow");
            return 0;
        }
        else{
            this.top++;
            this.array[this.top]=value;
            console.log(value," is added in stack");
        }
    }

    pop(){
        if(this.top==-1){
            console.log("stack is underflow");
        }
        else{
            console.log(this.array[this.top], " is poped");
            this.top--;
        }
    }

    display(){
        if(this.top==-1){
            console.log("stack is Empty!");
        }
        else{
            for(var i=this.top; i>=0;i--){
                console.log(this.array[i]);
            }
        }
    }
}

var st = new stack();
console.log("1: push \n2: pop \n3: display\n4: exit");
while(true){
    var x=input("enter choice: ");
    if(x==1){
        var value = Number(input("enter value: "));
        st.push(value);
    }
    else if(x==2){
        st.pop();
    }
    else if(x==3){
        st.display();
    }
    else{
        break;
    }
}

File: Stack/stack.cpp
#include<iostream>
using namespace std;

struct stack{int size,top,*S;};


void creatstack(struct stack *st){
	cout<<"\nenter the size of stack\n";
	cin>>st->size;
	st->top=-1;
	st->S=new int[st->size];
}


void push(struct stack *st,int value){
	if(st->top==st->size-1){
		cout<<"\nstack overflow \n";
	}
	else{
		st->top++;
		st->S[st->top]=value;
		cout<<"\n"<<value<<" is added in stack\n";
	}
}

int pop(struct stack *st){
	int x=-1;
	if(st->top==-1){
		cout<<"\n stack is underflow\n";
	}
	else{
		x=st->S[st->top];
		st->top--;
		cout<<"\n"<<x<<" is removed from stack\n";
	}
	return x;
}
void display(struct stack *st){
	for(int i=st->top;i>=0;i--){
		cout<<"\n"<<st->S[i];
	}
}


int main(){
	struct stack st;
	creatstack(&st);
	while(true){
		cout<<"\npress 1 for push\npress 2 for pop\npress 3 for display\npress 4 for exit\nenter choice: ";
		int i;
		cin>>i;
		if (i==1)
		{
			int value;
			cout<<"enter the value: ";
			cin>>value;
			push(&st,value);
		}
		else if (i==2)
		{
			pop(&st);
		}
		else if (i==3)
		{
			display(&st);
		}
		else
		{
			break;
		}
	}

}









